shell/platform/windows/fixtures/main.dart

```diff
@@ -33,3 +35,28 @@ void verifyNativeFunctionWithReturn() {
   bool value = signalBoolReturn();
   signalBoolValue(value);
 }
+
+@pragma('vm:entry-point')
+void drawHelloWorld() {
+  ui.PlatformDispatcher.instance.onBeginFrame = (Duration duration) {
+    final ui.ParagraphBuilder paragraphBuilder = ui.ParagraphBuilder(ui.ParagraphStyle())
+      ..addText('Hello world');
+    final ui.Paragraph paragraph = paragraphBuilder.build();
+
+    paragraph.layout(const ui.ParagraphConstraints(width: 800.0));
+
+    final ui.PictureRecorder recorder = ui.PictureRecorder();
+    final ui.Canvas canvas = ui.Canvas(recorder);
+
+    canvas.drawParagraph(paragraph, ui.Offset.zero);
+
+    final ui.Picture picture = recorder.endRecording();
+    final ui.SceneBuilder sceneBuilder = ui.SceneBuilder()
+      ..addPicture(ui.Offset.zero, picture)
+      ..pop();
+
+    ui.window.render(sceneBuilder.build());
```

Comment:
I think you'll need a signal after the render call to guarantee that it actually runs -- fixtures are not guaranteed to run to completion, so you'll want a wait/signal pair if you need to guarantee execution to a certain point. Otherwise I suspect this could end up being flaky.

Reply:
From my understanding, this scenario would happen if the UI task runner was stopped, which happens when the engine is destroyed. However, I don't destroy the engine until after the next frame callback has been executed. This makes me think this won't be flaky (though it's very possible I've missed something!)

It seems this is a similar pattern to the embedder's unit tests; their entry points don't wait for a native callback. For example: https://github.com/flutter/engine/blob/5c60785b802ad2c8b8899608d949342d5c624952/shell/platform/embedder/fixtures/main.dart#L1056-L1073

Reply:
Yep, my bad - you're absolutely right; you're already hanging the test waiting on a latch till the callback is called, so we won't hit the EnginePtr destructor till after the latch is signalled, and therefore continue running the Dart side till then.